python/src/trezorlib/cli/firmware.py

```diff
@@ -183,6 +183,46 @@ def get_all_firmware_releases(
     return releases
 
 
+def get_url_and_fingerprint_from_release(
+    release: dict,
+    bitcoin_only: bool = False,
+) -> tuple:
+    """Get appropriate url and fingerprint from release dictionary."""
+    if bitcoin_only:
+        url = release["url_bitcoinonly"]
+        fingerprint = release["fingerprint_bitcoinonly"]
+    else:
+        url = release["url"]
+        fingerprint = release["fingerprint"]
+
+    url_prefix = "data/"
+    if not url.startswith(url_prefix):
+        click.echo(f"Unsupported URL found: {url}")
+        sys.exit(1)
+    final_url = "https://data.trezor.io/" + url[len(url_prefix) :]
+
+    return final_url, fingerprint
+
+
+def find_specified_firmware_version(
+    version: str,
+    beta: bool = False,
```

Comment:
ISTM these shouldn't be optional either, nor the arguments of `find_best_firmware_version`

are there any calls to these functions that omit these arguments?